pandas/core/array_algos/masked_reductions.py

```diff
@@ -63,17 +68,13 @@ def _minmax(func, values: np.ndarray, mask: np.ndarray, skipna: bool = True):
         Whether to skip NA.
     """
     if not skipna:
-        if mask.any():
+        if mask.any() or not values.size:
+            # min/max with empty array raise in numpy, pandas returns NA
             return libmissing.NA
         else:
-            if values.size:
-                return func(values)
-            else:
-                # min/max with empty array raise in numpy, pandas returns NA
-                return libmissing.NA
+            return func(values)
     else:
-        subset = values[~mask]
-        if subset.size:
+        if not mask.all():
```

Comment:
Doing a `mask.all()` might on average be expensive here, since in most cases you will actually need the subset afterwards

Reply:
Ah yes, I see what you're saying. I'll switch that back